File: admin/frontend/src/mock/tax.ts
import type { TaxRule } from "@/lib/types";

const now = "2026-07-20T10:00:00.000Z";

export const mockTaxRules: TaxRule[] = [
  {
    id: 1,
    name: "GST Standard",
    code: "GST18",
    rate_percent: 18,
    is_inclusive: false,
    country: "India",
    state: null,
    is_active: true,
    sort_order: 0,
    created_at: now,
    updated_at: now,
  },
  {
    id: 2,
    name: "GST Reduced",
    code: "GST5",
    rate_percent: 5,
    is_inclusive: false,
    country: "India",
    state: null,
    is_active: true,
    sort_order: 1,
    created_at: now,
    updated_at: now,
  },
  {
    id: 3,
    name: "Tamil Nadu SGST",
    code: "TN-SGST",
    rate_percent: 9,
    is_inclusive: true,
    country: "India",
    state: "Tamil Nadu",
    is_active: false,
    sort_order: 2,
    created_at: now,
    updated_at: now,
  },
];

File: backend/app/modules/settings/models/tax.py
from __future__ import annotations

import uuid
from datetime import datetime
from decimal import Decimal

from sqlalchemy import Boolean, DateTime, Integer, Numeric, String, Uuid, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class TaxRule(Base):
    __tablename__ = "tax_rules"

    id: Mapped[uuid.UUID] = mapped_column(
        Uuid(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    name: Mapped[str] = mapped_column(String(120))
    code: Mapped[str] = mapped_column(String(40), unique=True, index=True)
    rate_percent: Mapped[Decimal] = mapped_column(Numeric(5, 2))
    is_inclusive: Mapped[bool] = mapped_column(Boolean, default=False)
    country: Mapped[str | None] = mapped_column(String(100), nullable=True)
    state: Mapped[str | None] = mapped_column(String(100), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now()
    )
